src/home_robot/home_robot/utils/bullet.py

```diff
@@ -507,10 +521,15 @@ def compute_ik(
                             print("IK failure:")
                             print(q_out > self.range[:, 0])
                             print(q_out < self.range[:, 1])
-                        q_out = None
+                        success = False
             else:
                 q_out = q_full
-            if q_out is not None:
+
+            if success:
                 break
 
-        return q_out
+        debug_info = {"best_q_out": q_out}
+        if not success:
+            q_out = None
```

Comment:
Doing this for consistency, in case callers are expecting it to be None if it failed. Technically redundant though.